web/modules/custom/dpl_event/src/PriceFormatter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\dpl_event;
+
+use Brick\Math\BigDecimal;
+use Drupal\Core\StringTranslation\TranslationInterface;
+
+/**
+ * Formats prices according to local rules.
+ *
+ * This is important as we do not rely on fields built for prices. Instead we
+ * have double fields with values as strings.
+ */
+class PriceFormatter {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    protected TranslationInterface $translation,
+  ) {}
+
+  /**
+   * Format a single price.
+   */
+  public function formatPrice(string $price_string): string {
+    $context = ['context' => 'dpl_event'];
+
+    $price = BigDecimal::of($price_string);
+    return match(TRUE) {
```

Comment:
Uh..? Why do you use `match` with a hardcoded value?

Reply:
Because `match` needs a boolean input but [you can still use it for non-identity checks](https://www.php.net/manual/en/control-structures.match.php#weakreference.basic-example). It is basically another way of doing `if+elseif+else`.

Reply:
To be completely honest, I feel like this is a very complicated way of doing something, that is very hard to read/maintain.

Maybe it's just cause I've never used that function before, but it seems super abstract.
I get that it might be less code than a more simple if/else, but is it worth it?

I'll leave the decision up to you - I'll approve regardless :)